### PR TITLE
[fix](delete) Fix delete stmt on MOW table doesn't use partial update in Nereids planner

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

1. Fix nereids planner don't use partial update for delete statement on merge-on-write table introduced in https://github.com/apache/doris/pull/36782
2. add `IS_PARTIAL_UPDATE: true` in `OlapTableSink.getExplainString()`
3. don't convert the delete stmt to partial update if the table has cluster key.

branch-2.1-pick: https://github.com/apache/doris/pull/39214

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run p0

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run cloud_p0

File: regression-test/suites/delete_p0/test_delete_predicate_on_value.groovy [hunk 1]
```diff
@@ -134,23 +135,17 @@ suite("test_delete_on_value") {
     // test mor table
     sql "insert into ${tableName4} values(1,1,10);"
     sql "insert into ${tableName4} values(1,1,5);"
-    qt_sql "select * from ${tableName4} order by x,y,z;"
-    sql "set skip_storage_engine_merge=true;"
```
[the remark this answers is not in the document]
Comment (bobhan1) [reply regression-test/suites/delete_p0/test_delete_predicate_on_value.groovy:138]: it's case irrelevant

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**